circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -293,22 +293,15 @@ object CirceInstances {
       case None => Pull.output(emptyArray)
       case Some((hd, tl)) =>
         Pull.output(
-          Chunk.concat(Vector(CirceInstances.openBrace, fromJsonToChunk(printer)(hd)))
+          CirceInstances.openBrace ++ fromJsonToChunk(printer)(hd)
         ) >> // Output First Json As Chunk with leading `[`
-          tl.repeatPull {
-            _.uncons.flatMap {
-              case None => Pull.pure(None)
-              case Some((hd, tl)) =>
-                val interspersed = {
-                  val bldr = Chunk.newBuilder[Byte]
-                  hd.foreach { o =>
-                    bldr += CirceInstances.comma
-                    bldr += fromJsonToChunk(printer)(o)
-                  }
-                  bldr.result
-                }
-                Pull.output(interspersed) >> Pull.pure(Some(tl))
+          tl.mapChunks { c =>
+            val bldr = Chunk.newBuilder[Byte]
+            c.foreach { o =>
+              bldr += CirceInstances.comma
+              bldr += fromJsonToChunk(printer)(o)
             }
+            bldr.result
```

Comment:
Perhaps this can be a local `def nonFirst(c: Chunk[Json]): Chunk[Byte]` function?

Reply:
Additionally, rather than using the `fromJsonToChunk` on every element, and using chunk concatenation (in the bldr builder), would it also help to use one bytebuffer per source chunk, directly print each json to that buffer, and use one view ?

Reply:
> Additionally, rather than using the fromJsonToChunk on every element, and using chunk concatenation (in the bldr builder), would it also help to use one bytebuffer per source chunk, directly print each json to that buffer, and use one view ?

Benchmarking `streamJsonArrayEncoder` and `jsonEncoder` makes it clear that there is a lot of room for improvement. That feels a bit out of scope of my small refactoring in this PR.